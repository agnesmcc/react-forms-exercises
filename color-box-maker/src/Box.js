import React from "react";

const Box = ({id, color, width, height, removeBox}) => {
    return (
        <div style={{
            backgroundColor: color, 
            width: width, 
            height: height
        }}>
            <button onClick={() => removeBox(id)}>X</button>
        </div>
    )
}

export default Box;
