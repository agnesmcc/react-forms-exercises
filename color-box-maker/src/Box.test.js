import { render } from '@testing-library/react';
import Box from './Box';

const BOX = (
    <div data-testid="test-box">
    <Box
        id={1}
        width={100}
        height={100}
        color="red"
    />
    </div>
)
    

it('renders without crashing', () => {
  const { getByTestId } = render(BOX);
  const box = getByTestId('test-box');
  const child = box.firstChild;
  expect(child).toHaveStyle('background-color: red');
})

it('matches the snapshot', () => {
  const { asFragment } = render(BOX);
  expect(asFragment()).toMatchSnapshot();
}) 
