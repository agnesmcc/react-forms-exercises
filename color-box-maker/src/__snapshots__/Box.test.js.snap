// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="test-box"
  >
    <div
      style="background-color: red; width: 100px; height: 100px;"
    >
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
